c_src/xav/reader.c

```diff
@@ -175,6 +184,24 @@ int reader_next_frame(struct Reader *reader) {
   return 0;
 }
 
+int reader_seek_to_time(struct Reader *reader, double time_in_seconds) {
+    int64_t frmseekPos = av_rescale_q(
+        (int64_t)(time_in_seconds * AV_TIME_BASE),
+        AV_TIME_BASE_Q,
+        reader->fmt_ctx->streams[reader->stream_idx]->time_base
+    );
+
+    avcodec_flush_buffers(reader->c);
+
+    if (av_seek_frame(reader->fmt_ctx, reader->stream_idx, frmseekPos, AVSEEK_FLAG_BACKWARD) < 0) {
```

Comment:
In my test video, there is a keyframe every second with a framerate of 25 fps. But when i seek, i can only get to frames which are a multiple of 250 frames...couldn't figure out where the issue is. Maybe we have to explicitly calculate P frames?

Reply:
That was indeed the issue! `av_seek_frame` does only seek to an I frame and you have to send packets to the decoder until you reach the desired timestamp. After some checking i confirmed that my test file had a keyframe every 10 seconds instead of every 1 seconds